daemon/daemon.go

```diff
@@ -526,7 +526,7 @@ func (daemon *Daemon) restore() error {
 			if err := daemon.prepareMountPoints(c); err != nil {
 				log.WithError(err).Error("failed to prepare mount points for container")
 			}
-			if err := daemon.containerStart(c, "", "", true); err != nil {
+			if err := daemon.containerStart(ctx, c, "", "", true); err != nil {
```

Comment:
Good catch. There are many other usages of `context.Background()` in this function (mostly semaphore.Acquire though). I think plumbing context without addressing all other usages isn't really useful. Since passing context here is not really crucial for unblocking us, for now I'll just use `context.Background()` here like anything else that accepts context in this function. Hopefully we'll be able to get back to this on future, maybe when refactoring this function completely (it's definitely too big). WDYT?

Reply:
[I agree, 100%.](https://github.com/moby/moby/pull/43401#discussion_r842918719)